### PR TITLE
[core][job] Spammy get job info logs 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
image

With the dashboard/infra keep getting jobs info, this logging is too spammy making the rest of GCS logs not visible. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]

File: src/ray/gcs/gcs_server/gcs_job_manager.cc [hunk 1]
```diff
@@ -146,7 +146,7 @@ void GcsJobManager::AddJobFinishedListener(JobFinishListenerCallback listener) {
 void GcsJobManager::HandleGetAllJobInfo(rpc::GetAllJobInfoRequest request,
                                         rpc::GetAllJobInfoReply *reply,
                                         rpc::SendReplyCallback send_reply_callback) {
-  RAY_LOG(INFO) << "Getting all job info.";
+  RAY_LOG(DEBUG) << "Getting all job info.";
```
Comment (vitsai) [inline src/ray/gcs/gcs_server/gcs_job_manager.cc:149]: Not sure whether RAY_LOG(DEBUG) or RAY_LOG_EVERY_N(INFO) makes more sense, but no strong opinion

**Review by vitsai — APPROVED** [review]
cc: @iycheng as well

File: src/ray/gcs/gcs_server/gcs_job_manager.cc (hunk 1 above)
Comment (rickyyx) [reply src/ray/gcs/gcs_server/gcs_job_manager.cc:149]: `RAY_LOG_EVERY_N(INFO)` yeah, thought about this as well. But I feel occasional logging is probably also not useful for this per-request happy path.

Comment (rkooo567) [reply src/ray/gcs/gcs_server/gcs_job_manager.cc:149]: maybe just remove all logs? Seems not that useful in general

**Comment by rickyyx** [comment]
Tests failures unrelated

**Review by fishbone — APPROVED** [review]

**Comment by iycheng** [comment]
LGTM. But do you know why there are so many RPC requets for this?

**Comment by rickyyx** [comment]
> LGTM. But do you know why there are so many RPC requets for this?

I think we are periodically checking for job status when dashboard is open?

**Comment by rkooo567** [comment]
This is used to detect idleness of the cluster

**Comment by rickyyx** [comment]
> This is used to detect idleness of the cluster

oh interesting - do you know the caller? we should probably use the idleness info now available in the GCS for this.

**Comment by rickyyx** [comment]
Test failures unrelated.

**Pull request merged**